core-interfaces/ICreateSessionOptions.ts

```diff
@@ -11,4 +12,5 @@ export default interface ICreateSessionOptions extends ISessionOptions {
   viewport?: IViewport;
   timezoneId?: string;
   locale?: string;
+  upstreamProxy?: IUpstreamProxy;
```

Comment:
should this be `upstreamProxyUrl`?

Reply:
Oh, never mind... sorry... I see 🙃

Reply:
Well, opening this line of questing back up... do you think it would be simpler to just specify a `upstreamProxyURL` and let the `auth` (if needed) be in the URL?

Reply:
The proxy configurations on the vpn providers tend to have separated url and auth, so I just thought this was a bit easier to translate without having to go make sure you've got formatting correct. This approach allows both. I previously set it up as separate properties (not an object), but then through there might be situations where want want additional configs on the proxy - like, bypassForDnsLookups, for an example. Does this lean you either way?

Reply:
What if the URL includes auth details + auth key/value is include?

Reply:
A simple upstreamProxyUrl just seems "simpler" than an object... I guess it seems that way to me since Chrome accept a string...

Reply:
I was trying to make things easier, but if it's not easier, totally fine dropping it. The code will take auth precedence from the "auth" field over the url, but will take from either.

Reply:
Removed.